pkg/strategy/tradingdesk/manager.go

```diff
@@ -0,0 +1,207 @@
+	if riskPerUnit.Sign() <= 0 {
+		return fixedpoint.Zero, m.createInvalidStopLossError(param.Side, currentPrice)
+	}
+
+	availableBalance, err := m.getAvailableBalance(param.Side)
+	if err != nil {
+		return fixedpoint.Zero, err
+	}
+
+	// Calculate maximum quantity based on MaxLossLimit
+	var maxQuantityByRisk fixedpoint.Value
+	if !m.MaxLossLimit.IsZero() {
+		maxQuantityByRisk = m.MaxLossLimit.Div(riskPerUnit)
+	} else {
+		maxQuantityByRisk = param.Quantity
+	}
+
+	// Calculate maximum quantity based on available balance
+	var maxQuantityByBalance fixedpoint.Value
+	if param.Side == types.SideTypeBuy {
+		// For buy orders: available quote currency / price
+		maxQuantityByBalance = availableBalance.Div(currentPrice)
+	} else {
+		// For sell orders: available base currency
+		maxQuantityByBalance = availableBalance
+	}
+
+	// Use the minimum of the three: original quantity, risk-based limit, balance-based limit
+	quantity := fixedpoint.Min(param.Quantity, fixedpoint.Min(maxQuantityByRisk, maxQuantityByBalance))
+
+	// Ensure quantity is positive
+	if quantity.Sign() <= 0 {
+		return fixedpoint.Zero, fmt.Errorf("calculated quantity is zero or negative")
+	}
+
+	logrus.Infof("Position size calculation: symbol=%s, currentPrice=%s, stopLoss=%s, riskPerUnit=%s, maxLossLimit=%s, availableBalance=%s, finalQuantity=%s",
+		param.Symbol, currentPrice, param.StopPrice, riskPerUnit, m.MaxLossLimit, availableBalance, quantity)
+
+	return quantity, nil
+}
+
+// stopLossRange calculates the risk per unit based on current price, stop loss, and side
+func (m *TradingManager) stopLossRange(currentPrice, stopLossPrice fixedpoint.Value, side types.SideType) fixedpoint.Value {
+	if side == types.SideTypeBuy {
+		// For long positions, risk is current price - stop loss price
+		return currentPrice.Sub(stopLossPrice)
+	}
+	// For short positions, risk is stop loss price - current price
+	return stopLossPrice.Sub(currentPrice)
+}
+
+// createInvalidStopLossError creates an appropriate error message for invalid stop loss prices
+func (m *TradingManager) createInvalidStopLossError(side types.SideType, currentPrice fixedpoint.Value) error {
+	if side == types.SideTypeBuy {
+		return fmt.Errorf("invalid stop loss price for buy order: stop loss should be below current price (%s)", currentPrice.String())
+	}
+	return fmt.Errorf("invalid stop loss price for sell order: stop loss should be above current price (%s)", currentPrice.String())
+}
+
+// getAvailableBalance returns the available balance for the appropriate currency based on order side
+func (m *TradingManager) getAvailableBalance(side types.SideType) (fixedpoint.Value, error) {
+	account := m.Session.GetAccount()
+
+	var currency string
+	if side == types.SideTypeBuy {
+		currency = m.Market.QuoteCurrency // For buy orders, we need quote currency balance
+	} else {
+		currency = m.Market.BaseCurrency // For sell orders, we need base currency balance
```

Comment:
we don't use BaseCurrency for the futures account,

we only deposit USDT in the futures account, and there will be no base assets